k8s/tools/deploy-pubsub-gke.sh

```diff
@@ -0,0 +1,278 @@
+#!/bin/bash
+# Turbinia GKE deployment script
+# This script can be used to deploy the Turbinia stack to GKE PubSub. 
+# Requirements:
+# - have 'gcloud' and 'kubectl' installed.
+# - autheticate against your GCP project with "gcloud auth login"
+# - optionally have the GCP project set with "gcloud config set project [you-project-name]"
+#
+# Use --help to show you commands supported.
+
+set -o posix
+set -e
+
+# Source cluster config to pull specs to create cluster from. Please review
+# the config file and make any necessary changes prior to executing this script
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+source $DIR/.clusterconfig
+cd $DIR/..
+
+if [[ "$*" == *--help ]] ; then
+  echo "Turbinia deployment script for Kubernetes environment"
+  echo "Options:"
+  echo "--build-release-test           Deploy Turbinia release test docker image"
+  echo "--build-dev                    Deploy Turbinia development docker image"
+  echo "--no-gcloud-auth               Create service key instead of using gcloud authentication"
+  echo "--no-cloudfunctions            Do not deploy Turbinia Cloud Functions"
+  echo "--no-datastore                 Do not configure Turbinia Datastore"
+  echo "--no-filestore                 Do not deploy Turbinia Filestore"
+  echo "--no-gcs                       Do not create a GCS bucket"
+  echo "--no pubsub                    Do not create the PubSub and PSQ topic/subscription"
+  echo "--no-cluster                   Do not create the cluster"
+  exit 1
+fi
+
+# Check if gcloud is installed
+if [[ -z "$( which gcloud )" ]] ; then
+  echo "gcloud CLI not found.  Please follow the instructions at "
+  echo "https://cloud.google.com/sdk/docs/install to install the gcloud "
+  echo "package first."
+  exit 1
+fi
+
+# Check if kubectl is installed
+if [[ -z "$( which kubectl )" ]] ; then
+  echo "kubectl CLI not found.  Please follow the instructions at "
+  echo "https://kubernetes.io/docs/tasks/tools/ to install the kubectl "
+  echo "package first."
+  exit 1
+fi
+
+# Check configured gcloud project
+if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+  DEVSHELL_PROJECT_ID=$(gcloud config get-value project)
+  ERRMSG="ERROR: Could not get configured project. Please either restart "
+  ERRMSG+="Google Cloudshell, or set configured project with "
+  ERRMSG+="'gcloud config set project PROJECT' when running outside of Cloudshell."
+  if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+  echo "Environment variable \$DEVSHELL_PROJECT_ID was not set at start time "
+  echo "so attempting to get project config from gcloud config."
+  echo -n "Do you want to use $DEVSHELL_PROJECT_ID as the target project? (y / n) > "
+  read response
+  if [[ $response != "y" && $response != "Y" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+fi
+
+# Check if the configured VPC network exists.
+networks=$(gcloud -q compute networks list --filter="name=$VPC_NETWORK" |wc -l)
```

Comment:
Should these gcloud commands happen after the auth related bits below?

Reply:
moved auth part to before this gets run